tool/modules/uds_fuzz.py

```diff
@@ -0,0 +1,473 @@
+                                                session)
+                    if not Iso14229_1.is_positive_response(response):
+                        print("Unable to enter session. Retrying...\n")
+                        break
+
+                elif session_type[i] == "2" and session_type[i+1] == "7":
+                
+                    # Request seed
+                    session = str_to_hex(i, session_type)
+                    response = request_seed(arb_id_request, arb_id_response,
+                                            session, None, None)
+                    if response is None:
+                        print("\nInvalid response")
+                    elif Iso14229_1.is_positive_response(response):
+                        seed_list.append(list_to_hex_str(response[2:]))
+                        print("Seed received: {}\t(Total captured: {}, Delay used: {})"
+                            .format(list_to_hex_str(response[2:]),
+                                    len(seed_list),reset_delay), end="\r")
+
+                        if list_to_hex_str(response[2:]) == list_to_hex_str(str_to_int_list(target)):
+                            print("\n\nTarget seed found with delay: ", reset_delay)
+                            loop = False
+                            break
+
+                        stdout.flush()
+
+                    else:
+                        print_negative_response(response)
+                        break
+                
+                elif session_type[i] == 1 and session_type[i+1] == 1:
+                    ecu_reset(arb_id_request, arb_id_response, reset_type, None)
+                    time.sleep(reset_delay)
+                else:
+                    break
+
+
+            if reset_type:
+                ecu_reset(arb_id_request, arb_id_response, reset_type, None)
+                time.sleep(reset_delay)
+                reset_delay += 0.001
+
+    except KeyboardInterrupt:
+        print("Interrupted by user.")
+    except ValueError as e:
+        print(e)
+        return
+
+    if len(seed_list) > 0:
+        print("\n")
+        print("Security Access Seeds captured:")
+        for seed in seed_list:
+            print(seed)
+
+def str_to_hex(i, session_type):
+    session = []
+    session.append('0x')
+    session.append(session_type[i+2])
+    session.append(session_type[i+3])
+    session = ''.join(session)
+    session = int(session, 16)
+    return session
+
+
+def extended_session(arb_id_request, arb_id_response, session_type):
+    with IsoTp(arb_id_request=arb_id_request, arb_id_response=arb_id_response) as tp:
+        # Setup filter for incoming messages
+        tp.set_filter_single_arbitration_id(arb_id_response)
+        with Iso14229_1(tp) as uds:
+            response = uds.diagnostic_session_control(session_type)
+            return response
+
+
+def request_seed(arb_id_request, arb_id_response, level, data_record, timeout):
```

Comment:
This should be imported from the UDS module, rather than copy-pasted into this module.